lib/generators.nix

```diff
@@ -50,4 +50,18 @@ in
     in
       # map input to ini sections
       mapAttrsToStringsSep "\n" mkSection attrsOfAttrs;
+
+
+  /* Generates JSON from an arbitrary (non-function) value.
+    * For more information see the documentation of the builtin.
+    */
+  toJSON = {}: builtins.toJSON;
+
+
+  /* YAML has been a strict superset of JSON since 1.2, so we
+    * use toJSON. Before it only had a few differences referring
+    * to implicit typing rules, so it should work with older
+    * parsers as well.
+    */
+  toYAML = {}@args: toJSON args;
```

Comment:
So this covers the whole range of yam? Interesting.

I used the following snippet in the past: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/monitoring/telegraf.nix#L11